nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,818 @@
+"""High-level decoding object that exposes standard classification and
+regression strategies such as SVM, LogisticRegression and Ridge, with optional
+feature selection, and integrated hyper-parameter selection.
+"""
+# Authors : Yannick Schwartz
+#           Andres Hoyos-Idrobo
+#
+# License: simplified BSD
+
+import sklearn
+import itertools
+import warnings
+import numpy as np
+from distutils.version import LooseVersion
+from sklearn.externals.joblib import Parallel, delayed
+from sklearn.linear_model.ridge import Ridge, RidgeClassifier, _BaseRidge
+from sklearn.linear_model import LogisticRegression
+from sklearn.svm import LinearSVC, SVR
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.svm.bounds import l1_min_c
+from sklearn.base import RegressorMixin
+from sklearn.linear_model.base import LinearModel
+from sklearn.utils.extmath import safe_sparse_dot
+from sklearn import clone
+from sklearn.model_selection import ParameterGrid
+from sklearn.utils.validation import check_is_fitted, check_X_y
+from sklearn.metrics import check_scoring
+from sklearn.model_selection import check_cv
+
+from ..input_data.masker_validation import check_embedded_nifti_masker
+from .._utils.param_validation import _adjust_screening_percentile
+from .._utils.compat import _basestring
+from .._utils.cache_mixin import _check_memory
+from .._utils.param_validation import check_feature_screening
+from .._utils import CacheMixin
+
+
+SUPPORTED_ESTIMATORS = dict(
+    svc_l1=LinearSVC(penalty='l1', dual=False),
+    svc_l2=LinearSVC(penalty='l2'),
+    svc=LinearSVC(penalty='l2'),
+    logistic_l1=LogisticRegression(penalty='l1', solver='liblinear'),
+    logistic_l2=LogisticRegression(penalty='l2', solver='liblinear'),
+    logistic=LogisticRegression(penalty='l2'),
+    ridge_classifier=RidgeClassifier(),
+    ridge_regression=Ridge(),
+    ridge=Ridge(),
+    svr=SVR(kernel='linear'),
+)
+
+
+def _check_param_grid(estimator, X, y, param_grid):
+    """Check param_grid and return sensible default if none is given.
+    For more information, see the documentation:
+    <http://http://scikit-learn.org/stable/modules/generated/sklearn.svm.l1_min_c>
+    """
+    if param_grid is None:
+        param_grid = {}
+        # define loss function
+        if isinstance(estimator, LogisticRegression):
+            loss = 'log'
+        elif isinstance(estimator, (LinearSVC, _BaseRidge, SVR)):
```

Comment:
It should be Ridge or RidgeClassifier, no ?

Reply:
If I understand correctly in sklearn `Ridge` or `RidgeClassifier` is a wrapper class of `_BaseRidge`.